### PR TITLE
Add  LRU cache.

### PR DESCRIPTION
**Pull request by mk46**
Signed-off-by: Manish Kumar <manishjpiet@gmail.com>

Added LRU cache.
Ref : [envoyproxy/envoy/14341](https://github.com/envoyproxy/envoy/pull/14341)

### CODE REVIEW

**Comment by mk46** [comment]
CC @qiwzhang

File: BUILD [hunk 1]
```diff
@@ -18,11 +18,14 @@ cc_library(
         "jwt_verify_lib/check_audience.h",
         "jwt_verify_lib/jwks.h",
         "jwt_verify_lib/jwt.h",
+        "jwt_verify_lib/simple_lru_cache.h",
```
[the remark this answers is not in the document]
Comment (mk46) [reply BUILD]: Done.

**Review by nareddyt — APPROVED** [review]

File: simple_lru_cache/simple_lru_cache.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// simple_lru_cache_inl.h.
+
+#pragma once
+
+#include <functional>
+
+#include "absl/container/flat_hash_map.h"  // for hash<>
+
+namespace google {
+namespace jwt_verify {
```
Comment (qiwzhang) [inline simple_lru_cache/simple_lru_cache.h:23]: please change namespace from `jwt_verify` to `simple_lru_cache`  thanks

**Review by qiwzhang — CHANGES_REQUESTED** [review]

File: simple_lru_cache/simple_lru_cache.h (hunk 2 above)
Comment (mk46) [reply simple_lru_cache/simple_lru_cache.h]: Done.

**Review by qiwzhang — APPROVED** [review]

**Pull request merged**